### PR TITLE
Update data ingest project to focus entirely on inspection & upload of already validated documents

### PR DESCRIPTION
**Pull request by joel-wright**
This PR changes the intent of the data ingest from something that drives the API to create documents & instead uses the output from the new bulk-import endpoint.

The starting point is a single JSON file that contains an array of validated document objects. These objects are processed by:
 - downloading
 - creating an appropriate parser input depending on the result of hitting the source URL
 - uploading the downloaded document to a document cache bucket
 - creating individual parser input files in a second "pipeline bucket" that can be used to start parser work.

Still required:
 - implementation & integration with an endpoint to send md5sum & cdn urls to the API database (but this could be done as a post-processing stage)
 - the use of s3 metadata & parser input content to make a more refined decision about whether to overwrite & trigger further processing
 - tests (I know! I know! I'm sorry!!!)

### CODE REVIEW

**Comment by joel-wright** [comment]
@diversemix all the functional code now lives in `src/navigator_data_ingest/...` & all other `src` paths have been removed.

File: pyproject.toml [hunk 1]
```diff
@@ -46,13 +55,15 @@ asyncio_mode = "strict"
...
 D100,D103,D104,D107,
-D202,D203,D213,
+D202,D203,D212,
 D400,D401,D406,D407,D413,D415
 """
 
 [tool.flake8]
 max-line-length = 88
 extend-ignore = "E203"
+per-file-ignores = "__init__.py:F401"
```
Comment (diversemix) [inline pyproject.toml:65]: ooo - nice

File: src/navigator_data_ingest/base/api_client.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+"""A simple API client for creating documents & associations."""
+import hashlib
+import logging
+import os
+from functools import lru_cache
```
Comment (diversemix) [inline src/navigator_data_ingest/base/api_client.py:5]: always wanted an excuse to use this - nice!

File: src/navigator_data_ingest/base/api_client.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        return upload_result
+
+
+def _store_cached_document(
+    bucket: str,
+    name: str,
+    data: bytes,
+) -> str:
+    clean_name = name.lstrip("/")
+    output_file_location = S3Path(f"s3://{bucket}/{clean_name}")
```
Comment (diversemix) [inline src/navigator_data_ingest/base/api_client.py:139]: you can `S3Path("bucket", "folder", "file.txt") `

File: Dockerfile [hunk 4]
```diff
@@ -1,17 +1,17 @@
 FROM python:3.9
 
-RUN mkdir /app
-WORKDIR /app
+RUN mkdir /navigator-data-ingest
```
[the remark this answers is not in the document]
Comment (joel-wright) [reply Dockerfile:3]: I'm not sure that the name of the `WORKDIR` really matters, all I did was give it a nicer name...

File: src/navigator_data_ingest/base/api_client.py (hunk 3 above)
Comment (joel-wright) [reply src/navigator_data_ingest/base/api_client.py:139]: I'll do this

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (joel-wright) [reply Dockerfile:3]: have I missed something?

**Review by diversemix — APPROVED** [review]
All looking top notch sir!

**Comment by joel-wright** [comment]
> All looking top notch sir!

thanks for the review! onwards & upwards!

**Pull request merged**